### ISSUE TITLE
Hive does not recognise INTEGER type

### ISSUE THREAD

**Issue by juhoautio** [sqlfluff/sqlfluff#2296]
## Expected Behaviour

When placeholders are successfully configured, I expect both `fix` and `lint` to succeed.

## Observed Behaviour

I encountered a case where the presence of placeholders causes `lint` to fail with PRS error and exit code 65. For the same file `fix` succeeds with exit code 0.

## Steps to Reproduce

`lint_placeholders.sql`:
```sql
SELECT my_col
FROM my_schema.my_table
WHERE processdate = CAST(
        event_date AS INTEGER
    )
    AND eventtype = &{event_type}
```

```
$ sqlfluff fix -f lint_placeholders.sql
==== finding fixable violations ====
==== no fixable linting violations found ====
All Finished 📜 🎉!

$ echo $?
0
```

That is as expected.

```
$ sqlfluff lint lint_placeholders.sql
== [lint_placeholders.sql] FAIL
L:   4 | P:  70 |  PRS | Line 4, Position 70: Found unparsable section: ' AS
| INTEGER'
All Finished 📜 🎉!

$ echo $?
65
```

That is unexpected.

If I want to make `lint` pass regardless, I can add this in `.sqlfluff`:

```ini
[sqlfluff]
ignore = parsing
```

But I'm not sure how big the implications of setting that are. That conf option is not really documented or is it? Should it be?

## Dialect

Hive – but I don't know if this issue is dialect specific or not.

## Version

sqlfluff, version 0.9.1

Python 3.7.5

(not using dbt)

## Configuration

`.sqlfluff`:

```ini
[sqlfluff]
dialect = hive
templater = placeholder

[sqlfluff:templater:placeholder]
param_style = ampersand
event_type='my_event'
```

**Comment by tunetheweb** [comment]
This has nothing to do with the Placeholder templater. The same thing fails without any placeholders.

If you change `INTEGER` to `INT` it works.

This is because `INTEGER` is not set up in our Hive keywords file, only INT:
https://github.com/sqlfluff/sqlfluff/blob/7ffea4670c384f575dd7a59ed31da0b93ae8f800/src/sqlfluff/dialects/dialect_hive_keywords.py#L59

It looks to me from the Hive [source code](https://github.com/apache/hive/blob/bcc7df95824831a8d2f1524e4048dfc23ab98c19/ql/src/java/org/apache/hadoop/hive/ql/parse/HiveLexer.g#L125) that `INT` and `INTEGER` are equivalent. Is that the case?

In that case it should be simple enough to add this if you fancy opening a PR? Maybe with a test case to ensure it doesn't regress?

In meantime switching to `INT` will avoid this error.

**Comment by juhoautio** [comment]
Wow, thanks a lot! I somehow thought `lint` passed when I was not using any placeholder, but I made some mistake there. Sorry for a misleading description!

I can work on a PR to add INTEGER in hive dialect. I can also check if there are any other missing data type aliases.

**Comment by juhoautio** [comment]
As a side question in general: does it make sense that `fix -f` can return exit code 0 even though there is actually a PRS error happening? Why is it ignored by `fix -f` here?

**Comment by tunetheweb** [comment]
It probably doesn't. Could you raise a separate issue for that as probably should alert for it.

The reason it does is because it doesn't find anything to fix:

```
% sqlfluff fix --dialect hive test.sql  
==== finding fixable violations ====
==== no fixable linting violations found ====                                                                                                         
All Finished 📜 🎉!
```

It is however flagged as a linting error and if you assume `fix` is basically `lint` + `fix` (as many do if they `fix` in a pre-commit for example), then NOT alerting for this seems wrong.

I checked and unfixable errors (e.g. rule L007 which is. linting rule, but for which auto-fixing has not been implemented) ARE flagged. So to me it would make sense for consistency that other linting errors like this should be flagged by `fix`.

Anyway, if you raise that other issue we can have that discussion there, in case others disagree.

**Comment by juhoautio** [comment]
@tunetheweb thanks. I filed that as a new issue: "fix succeeds even though lint fails with PRS error", https://github.com/sqlfluff/sqlfluff/issues/2345.

This issue here remains simply about adding missing type aliases to Hive dialect.

**Comment by tunetheweb** [comment]
Fixed in #2296
